### PR TITLE
Add cryptocurrency nodes charts

### PR DESCRIPTION
**Pull request by daniel-yavorovich**
*****************************************************************************************

Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the 
history. This will make it easier to identify new changes. The PR will be squashed 
anyways when it is merged. Thanks.

*****************************************************************************************

### CODE REVIEW

File: incubator/bitcoincashd/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+- bitcoincashd
+- cryptocurrency
+- blockchain
+home: https://www.bitcoincash.org
+icon: https://www.bitcoincash.org/img/logo-dark.png
+sources:
+- https://github.com/kubernetes/charts
+- https://github.com/kuberstack/bitcoincashd
+maintainers:
+- name: Daniel Yavorovych
```
Comment (unguiculus) [inline incubator/bitcoincashd/Chart.yaml]: Use Github username and add `appVersion`.

File: incubator/bitcoincashd/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+[BitcoinCash](https://bitcoincash.org/) is Peer-to-Peer Electronic Cash. This is one of the Bitcoin forks.
+
+## Introduction
+
+This chart bootstraps a single node BitcoinCash deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
+
+## Prerequisites
+
+- Kubernetes 1.6+ with Beta APIs enabled
```
Comment (unguiculus) [inline incubator/bitcoincashd/README.md]: We support the current and previous minor versions, which would be 1.7+.

File: incubator/bitcoincashd/templates/configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{{- if .Values.configurationFiles }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "bitcoincashd.fullname" . }}
```
Comment (unguiculus) [inline incubator/bitcoincashd/templates/configmap.yaml:5]: Add standard labels.

File: incubator/bitcoincashd/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "bitcoincashd.fullname" . }}
+  labels:
+    app: {{ template "bitcoincashd.fullname" . }}
```
Comment (unguiculus) [inline incubator/bitcoincashd/templates/deployment.yaml]: `app` label should be `{{ template "bitcoincashd.name" . }}`.

File: incubator/bitcoincashd/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  template:
+    metadata:
+      labels:
+        app: {{ template "bitcoincashd.fullname" . }}
```
Comment (unguiculus) [inline incubator/bitcoincashd/templates/deployment.yaml]: Add `release` label.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Thanks for the PR. I'm made a few initial comments. Please read our review guidelines and follow our best practices. 

https://github.com/unguiculus/charts/blob/master/REVIEW_GUIDELINES.md

File: incubator/bitcoincashd/templates/configmap.yaml (hunk 3 above)
Comment (daniel-yavorovich) [reply incubator/bitcoincashd/templates/configmap.yaml:5]: Thank you! I have done all recommendations.

File: incubator/bitcoind/values.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
+image: "kuberstack/bitcoind"
+imageTag: "latest"
+imagePullPolicy: IfNotPresent
+
+RPCPort: 443
```
Comment (unguiculus) [inline incubator/bitcoind/values.yaml]: `rpcPort`

File: incubator/bitcoind/values.yaml [hunk 7]
```diff
@@ -0,0 +1,37 @@
+image: "kuberstack/bitcoind"
+imageTag: "latest"
+imagePullPolicy: IfNotPresent
```
Comment (unguiculus) [inline incubator/bitcoind/values.yaml]: Use nesting:

```yaml
image:
  repository:
  tag:
  pullPolicy:
```

File: incubator/bitcoind/values.yaml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  ##   set, choosing the default provisioner.  (gp2 on AWS, standard on
+  ##   GKE, AWS & OpenStack)
+  ##
+  # storageClass: "-"
+  accessMode: ReadWriteOnce
+  size: 300Gi
+
+## Configure resource requests and limits
+## ref: http://kubernetes.io/docs/user-guide/compute-resources/
+##
```
Comment (unguiculus) [inline incubator/bitcoind/values.yaml:27]: Comment out defaults:

```yaml
resources: {}
  # requests:
    # memory: 512Mi
    # cpu: 300m
```

File: incubator/geth/values.yaml [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+imageTag: "alpine"
+imagePullPolicy: IfNotPresent
+
+RPCPort: 8545
+WSPort: 8546
+RPCApi: "net,eth,personal,web3"
+WSApi: "net,eth,personal,web3"
+WSOrigins: "*"
+NetworkID: 1
+SyncMode: "fast"
```
Comment (unguiculus) [inline incubator/geth/values.yaml]: Start with lower case letter.

`rpcPort`, `wsApi`, `wsOrigins`, `networkId`, etc.

**Review by unguiculus — CHANGES_REQUESTED** [review]
I made some more comments. Please apply them to all charts.

**Comment by unguiculus** [comment]
Please use a consistent list indentation style. I'd suggest you always indent lists one level.

**Comment by daniel-yavorovich** [comment]
> Please use a consistent list indentation style. I'd suggest you always indent lists one level.

@unguiculus can you clarify this?

**Comment by unguiculus** [comment]
@daniel-yavorovich Here's two ways of expressing the same list in YAML:

```yaml
list:
- item1
- item2
- item2
```

```yaml
list:
  - item1
  - item2
  - item2
```

You use a mixture of both styles. I'd like you to be consistent throughout the chart, preferably indenting the list items.

**Comment by jackzampolin** [comment]
I'm very excited about these charts and could lend a hand if you need one @daniel-yavorovich!

**Comment by daniel-yavorovich** [comment]
@unguiculus I have changed list styles.

@jackzampolin thank you very mach! At the moment, no help is needed :)

**Comment by unguiculus** [comment]
Sorry if this comes a little late, but I'd really like you to split this up into seperate PRs per chart. Thanks.

File: incubator/geth/templates/NOTES.txt [hunk 10]
```diff
@@ -0,0 +1,18 @@
+geth RPC can be accessed via port {{ .Values.rpcPort }} on the following DNS name from within your cluster:
+{{ template "geth.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local
+
+To connect to geth RPC:
+
+1. Run an Ubuntu pod that you can use as a client:
+
+    kubectl run -i --tty ubuntu --image=ubuntu:16.04 --restart=Never -- bash -il
+
+2. Install the bitcoind client:
```
Comment (corymsmith) [inline incubator/geth/templates/NOTES.txt:10]: This should be `ethereum` client.

**Comment by corymsmith** [comment]
@daniel-yavorovich Is there a reason why you made these Deployments and not StatefulSets?

**Comment by daniel-yavorovich** [comment]
@corymsmith thanks, I will fix ethereum Notes

>  Is there a reason why you made these Deployments and not StatefulSets?

Deployments is lightweight. But scale using StatefulSets and call nodes by hostname may be good idea.

**Comment by jackzampolin** [comment]
One way to do this w/o stateful sets would be to enable labeling of the resources and then make the service optional with some config options. This way you could spin up new instances and once they are caught up to chain tip then you can modify deployment to add tag and include them in the service. Just a thought. That would require significantly less modification.

Also this chart (for `bitcoind`) has the RPC port in a non standard location (443 as opposed to 8333). I think moving to standard ports would be nice. Users can add SSL with something like `cert-manager`. I don't think the chart needs native support for that. Adding an ingress would be the right way to add SSL support to this chart.

I would love to see `zmq` support built into this chart as well. I think in many settings where you are going to be running `bitcoind` in kube you will want to get the notifications for something like a UTXO service.

**Comment by govale** [comment]
@unguiculus PR was splitted into separate PR's per chart:
https://github.com/kubernetes/charts/pull/3643
https://github.com/kubernetes/charts/pull/3644
https://github.com/kubernetes/charts/pull/3645
https://github.com/kubernetes/charts/pull/3646
https://github.com/kubernetes/charts/pull/3647
https://github.com/kubernetes/charts/pull/3648

**Comment by unguiculus** [comment]
Can this be closed? @daniel-yavorovich

**Comment by daniel-yavorovich** [comment]
@unguiculus yes, we can close this issue :)

**Pull request closed without merging**